test/common/http/conn_pool_grid_test.cc

```diff
@@ -35,27 +35,24 @@ namespace Http {
 class ConnectivityGridForTest : public ConnectivityGrid {
 public:
   using ConnectivityGrid::ConnectivityGrid;
+  using ConnectivityGrid::getOrCreateHttp2Pool;
+  using ConnectivityGrid::getOrCreateHttp3Pool;
 
   static bool hasHttp3FailedRecently(const ConnectivityGrid& grid) {
     return grid.getHttp3StatusTracker().hasHttp3FailedRecently();
   }
 
-  static ConnectionPool::Instance* forceCreateNextPool(ConnectivityGrid& grid) {
-    return grid.createNextPool();
+  static ConnectionPool::Instance* forceGetOrCreateHttp3Pool(ConnectivityGrid& grid) {
+    return grid.getOrCreateHttp3Pool();
+  }
+  static ConnectionPool::Instance* forceGetOrCreateHttp2Pool(ConnectivityGrid& grid) {
```

Comment:
nit: Since we're now doing "using ConnectivityGrid::getOrCreateHttp2Pool" do we need this helper function, or can we call grid.getOrCreateHttp2Pool() directly?

Reply:
by default the test grid creates mock pools.  This allows forcing a real grid's real pools.  commented inline